src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2816,11 +2817,16 @@ def create_function(cmd, resource_group_name, name, storage_account, plan=None,
                 site_config.app_settings.append(NameValuePair(name='WEBSITES_ENABLE_APP_SERVICE_STORAGE',
                                                               value='false'))
                 site_config.linux_fx_version = _format_fx_version(deployment_container_image_name)
+
+                # clear all runtime specific configs and settings
+                site_config_json = {KEYS.USE_32_BIT_WORKER_PROC: False}
+                app_settings_json = {}
```

Comment:
this seems dangrous, will user's app_settings be cleared also?

Reply:
Yeah sorry this is a little misleading. I'm not clearing anything from the functionapp definition (what we use in the request to make the function app).

These two variables (`site_config_json` and `app_settings_json`) only contain json from the stacks API, and represent what app settings and site configs we'll be setting  on the functionapp definition _later_ -- [here specifically](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L2826-L2833). Since linux custom image apps shouldn't be using the stacks api (they're providing their own image that contains these app settings instead of using our image and stacks api settings), we're just pre-emptively clearing these settings before they overwrite what's in the functionapp definition.

In a nutshell we're saying -- "Since you're providing your own image and settings, we'll go ahead and clear the settings we would normally set before we set them, especially since we don't even know what runtime your custom image is using."